okio/src/main/java/okio/Buffer.java

```diff
@@ -1283,9 +1292,9 @@ Segment writableSegment(int minimumCapacity) {
       if (s == null) {
         // No segments to scan!
         return -1L;
-      } else if (size - fromIndex < fromIndex) {
+      } else if (toIndex - fromIndex < fromIndex) {
         // We're scanning in the back half of this buffer. Find the segment starting at the back.
-        offset = size;
+        offset = toIndex;
         while (offset > fromIndex) {
```

Comment:
Rolled back to the original version -- we still need to find the segment that contains toIndex, because that's where the scan will start.